### DISCUSSION TITLE
Auto-generate HTML preview for GitHub Pages on push to GitHub repo

### DISCUSSION THREAD

**Discussion by sloede** [fonsp/Pluto.jl#2386] [Questions about using Pluto] [answered]
Is there a way to automatically generate the static HTML preview after pushing a Pluto notebook to a GitHub repo? Ideally I'd by looking for a GitHub Action that would generate the HTML preview for me such that I can show it via GH Pages.

As a bonus, is there a way to make Pluto download the notebook file using its original name instead of just `notebook.jl` when downloading from the HTML page?

**Comment by fonsp** [comment, marked as answer]
Hi @sloede, thanks for the question, and sorry for the late response!

Definitely! Take a look at https://github.com/JuliaPluto/static-export-template 🎉

**Reply by fonsp** [reply]
This will also solve your bonus question!
